modules/core/src/test/java/org/apache/ignite/internal/causality/IncrementalVersionedValueTest.java

```diff
@@ -341,6 +344,51 @@ void testCompleteMultipleFutures() {
         assertThat(future3, willBe(2));
     }
 
+    /**
+     * Tests that {@link IncrementalVersionedValue#dependingOn(IncrementalVersionedValue)} provides causality between 2 different values.
+     */
+    @RepeatedTest(100)
+    public void testDependingOn() {
+        var vv0 = new IncrementalVersionedValue<>(register, () -> 1);
+
+        var vv1 = new IncrementalVersionedValue<>(dependingOn(vv0), () -> 1);
+
+        int token = 1;
+
+        vv0.update(token, (i, e) -> completedFuture(i + 1));
+
+        vv1.update(token, (i, e) -> completedFuture(i + 1));
+
+        register.moveRevision(token);
```

Comment:
Maybe I'm looking in the wrong direction, but I don't see any concurrency here (no async calls), so everything will be executed in the same thread and the order of execution will be deterministic. Would it make sense to make the test more interesting by using something like `supplyAsync()` instead of `completedFuture()`?

Reply:
Ok, I'll do it